handlers/search.js

```diff
@@ -103,36 +96,47 @@ function getFilters(query) {
 
   const hasPublishedResults = query.has_published_results;
   if (hasPublishedResults) {
-    filters.has_published_results = (hasPublishedResults === 'true');
+    filters.has_published_results = hasPublishedResults;
   }
 
   return filters;
 }
 
+function validateQueryParams(query) {
+  const schema = Joi.object().keys({
+    page: Joi.number().integer().min(1).max(100),
+    registration_date_start: Joi.date().format('YYYY-MM-DD').empty('').raw(),
+    registration_date_end: Joi.date().format('YYYY-MM-DD').empty('').raw(),
+    location: Joi.array().single(true).items(Joi.string().empty('')),
+    q: Joi.string().empty(''),
+    problem: Joi.array().single(true).items(Joi.string().empty('')),
+    intervention: Joi.array().single(true).items(Joi.string().empty('')),
+    person: Joi.array().single(true).items(Joi.string().empty('')),
+    organisation: Joi.array().single(true).items(Joi.string().empty('')),
+    gender: Joi.valid(['male', 'female']).empty(''),
+    has_published_results: Joi.boolean().empty(''),
+    sample_size_start: Joi.number().integer().empty(''),
+    sample_size_end: Joi.number().integer().empty(''),
+  });
+
+  return Joi.validate(query, schema);
+}
+
 function searchPage(request, reply) {
-  const query = request.query;
+  const validatedQuery = validateQueryParams(request.query);
+  if (validatedQuery.error) {
+    reply(
+      Boom.badRequest('invalid query')
+    );
```

Comment:
Please, change this to `return reply(Boom.badRequest('Invalid query', validatedQuery.error)` (with the "Invalid" capitalized), so the rest of this function isn't executed.